src/nvim/option_defs.h

```diff
@@ -1080,4 +1081,24 @@ typedef struct vimoption {
 // buffers.  Indicate this by setting "var" to VAR_WIN.
 #define VAR_WIN ((char *)-1)
 
+// Option value type
+typedef enum {
+  kOptValTypeNil = 0,
+  kOptValTypeBoolean,
+  kOptValTypeNumber,
+  kOptValTypeString,
+} OptValType;
```

Comment:
Could name this `kOpt*` instead of `kOptVal*`, there is no risk of ambiguity in the future.

```c
typedef enum {
  kOptTypeNil = 0,
  kOptTypeBoolean,
  kOptTypeNumber,
  kOptTypeString,
} OptType;
```

Reply:
Won't this lead to ambiguity if an option can have multiple types in future?

Reply:
Agree with zeertz. Also, the current naming scheme for OptVal types is more consistent with other enums with types.